vscode/microsoft-kiota/src/steps.ts

```diff
@@ -154,41 +183,56 @@ export async function generateSteps(existingConfiguration: Partial<GenerateState
             ];
         }
     }
-    async function inputGenerationType(input: MultiStepInput, state: Partial<GenerateState>) {
-        const items = [l10n.t('Generate an API client'), l10n.t('Generate a plugin'), l10n.t('Generate an API manifest')];
-        const option = await input.showQuickPick({
-            title: l10n.t('What do you want to generate?'),
-            step: 1,
-            totalSteps: 1,
-            placeholder: l10n.t('Select an option'),
-            items: items.map(x => ({ label: x })),
-            validate: validateIsNotEmpty,
-            shouldResume: shouldResume
-        });
-        if (option.label === l10n.t('Generate an API client')) {
-            state.generationType = "client";
-            return (input: MultiStepInput) => inputClientClassName(input, state);
-        }
-        else if (option.label === l10n.t('Generate a plugin')) {
-            state.generationType = "plugin";
-            return (input: MultiStepInput) => inputPluginName(input, state);
-        }
-        else if (option.label === l10n.t('Generate an API manifest')) {
-            state.generationType = "apimanifest";
-            return (input: MultiStepInput) => inputManifestName(input, state);
+    function getNextStepForGenerationType(generationType: string|QuickPickItem) {
+        switch(generationType) {
+            case 'client':
+                return inputClientClassName;
+            case 'plugin':
+                return inputPluginName;
+            case 'apimanifest':
+                return inputManifestName;
+            default:
+                throw new Error('unknown generation type');
```

Comment:
Add the actual and expected value. L10N?

Reply:
I don't understand, this is not shown to the user. It's used in defining the next function to call. So doesn't need L10N translations.

Reply:
Error('unknown generation type') I mean